daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -362,13 +372,13 @@ object Node {
             signatories2,
             stakeholders2,
             key2,
-            _,
+            byKey2,
             ) =>
           import nf._
           coid === coid2 && templateId == templateId2 &&
           actingParties.forall(_ => actingParties == actingParties2) &&
           signatories == signatories2 && stakeholders == stakeholders2 &&
-          key.forall(_ => key == key2)
+          key.forall(_ => key == key2) && byKey.forall(_ => byKey == byKey2)
```

Comment:
This allows `key == None` and `key2 == Some(_)`. Is that intended? Same for `byKey`. If it is, I'd be more comfortable if we made it more explicit:
```suggestion
          (!key.isDefined() || key == key2) && (!byKey.isDefined() || byKey == byKey2)
```
Same below.